budget-app/budget.py

```diff
@@ -0,0 +1,115 @@
+from email import header
+from string import whitespace
+from xml.dom.minidom import CharacterData
+
+
+class Category:
+    def __init__(self, category):
+        self.ledger = []
+        self.amount = 0
+        self.category = category
+
+    # Deposit Function
+    def deposit(self, amount, desc=""):
+        appendList = {"amount": amount, "description": desc}
+        self.ledger.append(appendList)
+        self.amount += amount
+
+    # Withdraw Function
+    def withdraw(self, amount, desc=""):
+        if self.check_funds(amount) == True:
+            self.amount -= amount
+            appendList = {"amount": -amount, "description": desc}
+            self.ledger.append(appendList)
+            return True
+        else:
+            return False
+
+    # Balance Function
+    def get_balance(self):
+        return self.amount
+
+    # Check Funds Availability
+    def check_funds(self, amount):
+        if self.amount < amount:
+            return False
+        else:
+            return True
+
+    # Transfer Funds
+    def transfer(self, amount, category):
+        if self.check_funds(amount) == True:
+            self.amount -= amount
+            self.ledger.append({"amount": -amount, "description": "Transfer to " + category.category})
+            category.deposit(amount, "Transfer from " + self.category)
+            return True
+        else:
+            return False
+
+    # String Display
+    def __str__(self):
+        category_len = len(self.category)
+        max_len = 30
+        asteriks = []
+
+        for i in range(max_len - category_len):
+            asteriks.append('*')
+        oneSideAsteriks = int(len(asteriks)/2)
+        asteriks.insert(oneSideAsteriks, self.category)
+        header = "".join(asteriks)
+        ledger_items = []
+        for item in self.ledger:
+            whitespace = max_len
```

Comment:
No need of `whitespace` and white spaces can be handled with `rjust` and `ljust`